examples/rdkafka_complex_consumer_example.cpp

```diff
@@ -81,7 +81,7 @@ static void print_time () {
         struct timeval tv;
         char buf[64];
         gettimeofday(&tv, NULL);
-        strftime(buf, sizeof(buf) - 1, "%Y-%m-%d %H:%M:%S", localtime(&tv.tv_sec));
+        strftime(buf, sizeof(buf) - 1, "%Y-%m-%d %H:%M:%S", localtime((time_t *)&tv.tv_sec));
```

Comment:
Not sure this cast is safe.
When (on what platform) is tv_sec not a time_t?

Reply:
Looks like winsock2 pulls in this definition: https://github.com/mirror/mingw-w64/blob/master/mingw-w64-headers/crt/_timeval.h#L12

https://developercommunity.visualstudio.com/content/problem/291705/move-timeval-struct-out-of-winsock2h.html

I will see if there's any way to avoid including winsock2 in this example. The danger would be if `sizeof(time_t) > sizeof(long)`.

Reply:
I opted to just replace the use of ATL with a WinAPI call here so that MinGW can share the code with MSVC. It wasn't apparent to me how I could stop winsock2.h getting included since it's needed for sockets.